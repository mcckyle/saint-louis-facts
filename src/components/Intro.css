/* Filename: Intro.css */

/* General styling for the Intro section */
.intro
{
    background: linear-gradient(130deg, #fdf7e4, #f9eccc);
    padding: 4.5rem 2rem;
    text-align: center;
    border-radius: 20px;
    box-shadow: 0 16px 48px rgba(0, 0, 0, 0.2);
    margin: 4rem auto;
    max-width: 1000px;
    font-family: 'Playfair Display', 'Garamond', serif;
    border: 4px double #d4af37; /* Gold border. */
    position: relative;
    transition: transform 0.4s ease, box-shadow 0.4s ease;

    /*background-color: rgba(253, 246, 227, 0.95);*/
}

.intro:hover
{
    transform: translateY(-6px);
    box-shadow: 0 22px 56px rgba(0, 0, 0, 0.25);
}

/* Title styling for the Intro section. */
.intro h2
{
  font-family: 'Cinzel Decorative', serif;
    font-size: 3.6rem;
    color: #5e2b00;
    margin-bottom: 1.6rem;
    text-transform: uppercase;
    letter-spacing: 3px;
    border-bottom: 4px double #d4af37;
    display: inline-block;
    padding-bottom: 0.7rem;
    text-shadow: 1.5px 1.5px #d4af37; /* Gold shadow. */
}

/* Wrapper for Intro section paragraph. */
.intro-text
{
    max-width: 820px;
    margin: 0 auto;
}

/* Paragraph styling. */
.intro p
{
    font-size: 1.4rem;
    line-height: 1.95;
    font-family: 'Garamond', serif;
    color: #2d2d2d;
    padding: 1.5rem 1.8rem;
    background-color: rgba(255, 255, 255, 0.88);
    border-radius: 12px;
    margin-top: 1.4rem;
    box-shadow: 0 4px 16px rgba(212, 175, 55, 0.18);
    border-left: 6px solid #bfa437;
    border-right: 6px solid #bfa437;
}

/* Responsiveness. */
@media (max-width: 768px) {
  .intro {
    padding: 3rem 1.2rem;
  }

  .intro-title {
    font-size: 2.5rem;
    letter-spacing: 2px;
  }

  .intro p {
    font-size: 1.15rem;
    padding: 1rem 1.2rem;
  }
}
