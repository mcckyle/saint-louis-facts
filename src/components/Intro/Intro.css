/* Filename: Intro.css */

/* ========================
   Intro Section Styling.
=========================== */
.intro
{
    position: relative;
    width: 100%;
    padding: 6rem 2rem;
    background: linear-gradient(160deg, #fffdf9, #f8f5eb);
    border-top: 4px solid var(--color-accent);
    text-align: center;
    overflow: hidden;
    animation: fadeInIntro 1s ease forwards;
}

/* Soft Golden Radial Glow Behind the Section. */
.intro::before
{
    content: '';
    position: absolute;
    top: -50%;
    left: -50%;
    width: 200%;
    height: 200%;
    background: radial-gradient(circle, rgba(212, 175, 55, 0.06), transparent 70%);
    z-index: 0;
}

/* Intro Title Styling. */
.intro-title
{
    font-family: 'Cinzel Decorative', serif;
    font-size: clamp(2.8rem, 5vw, 4rem);
    margin-bottom: 1.8rem;
    text-transform: uppercase;
    letter-spacing: 2px;
    background: linear-gradient(45deg, #b8860b, #d4af37, #ffecb3);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    position: relative;
}

/* Shimmer Effect on Underline. */
.intro-title::after
{
    content: '';
    position: absolute;
    bottom: -8px;
    left: 50%;
    transform: translateX(-50%);
    width: 60%;
    height: 3px;
    background: linear-gradient(to right, #d4af37, #fff4c2, #d4af37);
    background-size: 200% auto;
    animation: shimmer 5s linear infinite;
}

/* Lead Text Styling. */
.intro-lead
{
    max-width: 800px;
    margin: 0 auto;
    z-index: 1;
    position: relative;
}

/* Intro Paragraph Styling. */
.intro p
{
    font-size: 1.4rem;
    line-height: 1.9;
    font-family: 'Lora', serif;
    color: #3a3a3a;
    margin: 1.2rem auto 0;
    padding: 1.2rem 1.5rem;
    background: rgba(255, 255, 255, 0.85);
    border-radius: 12px;
    box-shadow: 0 6px 18px rgba(0, 0, 0, 0.05);
    transition: transform 0.3s ease, box-shadow 0.3s ease;
}

.intro p:hover
{
    transform: translateY(-2px);
    box-shadow: 0 8px 22px rgba(0, 0, 0, 0.08);
}

/* Scroll Cue Styles. */
.intro-scroll-cue
{
    width: 26px;
    height: 46px;
    border: 2px solid var(--color-accent);
    border-radius: 20px;
    position: relative;
    margin: 2rem auto 0;
    box-shadow: 0 0 6px rgba(212, 175, 55, 0.5);
}

.intro-scroll-cue::before
{
    content: '';
    position: absolute;
    top: 8px;
    left: 50%;
    width: 4px;
    height: 4px;
    background: var(--color-accent);
    border-radius: 50%;
    transform: translateX(-50%);
    animation: scrollDot 1.5s infinite;
}

/* Animations Styles. */
@keyframes shimmer
{
    0% { background-position: 0% center; }
    100% { background-position: 200% center; }
}

@keyframes fadeInIntro {
    from { opacity: 0; transform: translateY(30px); }
    to { opacity: 1; transform: translateY(0); }
}

@keyframes scrollDot
{
    0% { transform: translate(-50%, 0); opacity: 1; }
    50% { transform: translate(-50%, 14px); opacity: 0.6; }
    100% { transform: translate(-50%, 0); opacity: 1; }
}

/* Responsive Design Styles. */
@media (max-width: 768px) {
  .intro {
    padding: 3rem 1rem;
  }

  .intro-title {
    font-size: 2.2rem;
  }

  .intro p {
    font-size: 1.1rem;
    padding: 1rem;
  }
}

/* *** Old styles past this line!! ****/

/*.intro::after
 {  *
 content: '';
 position: absolute;
 top: 0;
 left: 0;
 width: 100%;
 height: 100%;
 opacity: 0.05;
 z-index: 0;
 animation: sparkle 20s linear infinite;
 pointer-events: none;
 }*/

/*.intro-title:hover
{
    text-shadow: 0 0 18px rgba(212, 175, 55, 0.6);
}*/

/*.intro p::before
 {  *
 content: '';
 position: absolute;
 top: 0;
 left: 0;
 width: 100%;
 height: 100%;
 background: linear-gradient(120deg, transparent, rgba(212, 175, 55, 0.06) transparent);
 z-index: -1;
 }*/

/*.intro-scroll-cue:hover
 {  *
 color: #ffd700;
 transform: scale(1.2);
 transition: transform 0.3s ease, color 0.3s ease;
 cursor: pointer;
 }*/

/*@keyframes bounce
 {
 0%, 100% { transform: translateY(0); }
 50% { transform: translateY(8px); }
 }*/

/*@keyframes sparkle
{
    from { background-position: 0 0; }
    to { background-position: 1000px 1000px; }
}

@keyframes pulseLine
{
    0%, 100% { transform: scaleY(1); opacity: 0.6; }
    50% { transform: scaleY(1.4); opacity: 1; }
}*/
