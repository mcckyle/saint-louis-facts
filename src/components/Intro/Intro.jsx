//Filename: Intro.jsx
//Author: Kyle McColgan
//Date: 13 October 2025
//Description: This file contains the intro component for the Saint Louis Facts React project Intro section.

import React from "react";
import "./Intro.css";

const Intro = React.forwardRef((props, ref) => (
    <section
      id="intro"
      ref={ref}
      className="intro"
      role="region"
      aria-labelledby="intro-heading"
    >
      <div className="intro-content">
        <h2 id="intro-heading" className="intro-title">
          Welcome to the Heart of <span className="highlight">Saint&nbsp;Louis</span>
        </h2>

        <p className="intro-text">
            Where the Mississippi meets a city of soul, creativity, and timeless character.
        </p>
        <p className="intro-text">
            From the Gateway Arch to its blues, baseball, and barbecue - explore the rhythm, resilience,
            and rich spirit that define the <span className="highlight">Gateway to the West</span>.
        </p>

        <div className="scroll-cue" aria-hidden="true" />
      </div>
    </section>
));

export default Intro;
