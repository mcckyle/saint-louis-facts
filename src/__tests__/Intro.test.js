//Filename: Intro.test.js
//Author: Kyle McColgan
//Date: 13 October 2025
//Description: This file contains the Jest unit tests for the Saint Louis Facts project Intro section.

import React from 'react';
import { render, screen } from '@testing-library/react';
import Intro from '../components/Intro/Intro.jsx';

describe('Intro Component', () => {
  beforeEach(() => {
    render(<Intro />);
  });

  test('renders the intro section', () => {
    const introSection = screen.getByRole('region');
    expect(introSection).toBeInTheDocument();
  });

  test('renders the correct heading', () => {
    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toHaveTextContent(/Welcome to the Heart of Saint Louis/i);
  });

  test('renders a paragraph with the correct description', () => {
    const paragraph = screen.getByText(/Where the Mississippi/i);
    expect(paragraph).toBeInTheDocument();
  });

  test('applies the correct class name to the intro section', () => {
    const introSection = screen.getByRole('region');
    expect(introSection).toHaveClass('intro');
  });

  test('contains exactly one heading element', () => {
    const headings = screen.getAllByRole('heading');
    expect(headings).toHaveLength(1);
  });

  test('contains exactly two paragraph elements', () => {
    const { container } = render(<Intro />);
    const paragraphs = container.querySelectorAll('p');
    expect(paragraphs).toHaveLength(2);
  });

  test('contains no additional content besides the heading and paragraph', () => {
    const sectionChildren = screen.getByRole('region').children;
    expect(sectionChildren).toHaveLength(1); // One intro content.
  });

  test('renders the intro section with correct structure', () => {
    const { container } = render(<Intro />);
    expect(container).toMatchSnapshot();
  });

  test('the paragraph text is visible and not empty', () => {
    const paragraph = screen.getByText(/Where the Mississippi/i);
    expect(paragraph).toBeVisible();
    expect(paragraph).not.toBeEmptyDOMElement();
  });

  test('renders an accessible region with appropriate role', () => {
    // render(<Intro />);
    const introSection = screen.getByRole('region');
    expect(introSection).toHaveAttribute('role', 'region');
  });

});
