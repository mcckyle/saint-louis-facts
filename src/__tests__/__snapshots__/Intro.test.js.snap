// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Component renders the intro section with correct structure 1`] = `
<div>
  <section
    aria-labelledby="intro-heading"
    class="intro"
    id="intro"
    role="region"
  >
    <div
      class="intro-content"
    >
      <h2
        class="intro-title"
        id="intro-heading"
      >
        Welcome to the Heart of 
        <span
          class="highlight"
        >
          SaintÂ Louis
        </span>
      </h2>
      <p
        class="intro-text"
      >
        Where the Mississippi meets a city of soul, creativity, and timeless character.
      </p>
      <p
        class="intro-text"
      >
        From the Gateway Arch to its blues, baseball, and barbecue - explore the rhythm, resilience, and rich spirit that define the 
        <span
          class="highlight"
        >
          Gateway to the West
        </span>
        .
      </p>
      <div
        aria-hidden="true"
        class="scroll-cue"
      />
    </div>
  </section>
</div>
`;
